tests/contrib/io/pyspark/test_spark_data_set.py

```diff
@@ -405,6 +411,70 @@ def test_prevent_overwrite(self, tmp_path, version, sample_spark_df):
             versioned_local.save(sample_spark_df)
 
 
+class TestSparkDataSetVersionedDBFS:
+    def test_load_latest(  # pylint: disable=too-many-arguments
+        self, mocker, versioned_dataset_dbfs, version, tmp_path, sample_spark_df
+    ):
+        mocked_glob = mocker.patch.object(versioned_dataset_dbfs, "_glob_function")
+        mocked_glob.return_value = [str(tmp_path / FILENAME / version.save / FILENAME)]
+
+        versioned_dataset_dbfs.save(sample_spark_df)
+        reloaded = versioned_dataset_dbfs.load()
+
+        assert mocked_glob.call_count == 2
+        mocked_glob.assert_has_calls(
+            [
+                call("/dbfs" + str(tmp_path / FILENAME / "*" / FILENAME)),
+                call("/dbfs" + str(tmp_path / FILENAME / "*" / FILENAME)),
+            ]
+        )
+        assert reloaded.exceptAll(sample_spark_df).count() == 0
+
+    def test_load_exact(self, tmp_path, sample_spark_df):
+        ts = generate_timestamp()
+        ds_dbfs = SparkDataSet(
+            filepath="/dbfs" + str(tmp_path / FILENAME), version=Version(ts, ts)
+        )
+
+        ds_dbfs.save(sample_spark_df)
+        reloaded = ds_dbfs.load()
+
+        assert reloaded.exceptAll(sample_spark_df).count() == 0
+
+    def test_save(  # pylint: disable=too-many-arguments
+        self, mocker, versioned_dataset_dbfs, version, tmp_path, sample_spark_df
+    ):
+        mocked_glob = mocker.patch.object(versioned_dataset_dbfs, "_glob_function")
+        mocked_glob.return_value = [str(tmp_path / FILENAME / version.save / FILENAME)]
+
+        versioned_dataset_dbfs.save(sample_spark_df)
+
+        mocked_glob.assert_called_once_with(
+            "/dbfs" + str(tmp_path / FILENAME / "*" / FILENAME)
+        )
+        assert (tmp_path / FILENAME / version.save / FILENAME).exists()
+
+    def test_exists(  # pylint: disable=too-many-arguments
+        self, mocker, versioned_dataset_dbfs, version, tmp_path, sample_spark_df
+    ):
+        mocked_glob = mocker.patch.object(versioned_dataset_dbfs, "_glob_function")
+        mocked_glob.return_value = [str(tmp_path / FILENAME / version.save / FILENAME)]
+
+        assert not versioned_dataset_dbfs.exists()
+
+        versioned_dataset_dbfs.save(sample_spark_df)
+        assert versioned_dataset_dbfs.exists()
+
+        assert mocked_glob.call_count == 3
+        mocked_glob.assert_has_calls(
```

Comment:
You could actually probably merge these two checks by doing sth like `assert mocked_glob.call_args_list == expected_calls` etc. But at the same time, why do you want to check the glob function calls here?

Reply:
The reason I'm checking glob calls is to make sure that version code is getting called for each `_load`/`_save`/`_exists` call... I think? Maybe you're right, it's an implementation detail that has no need to be tested (so what if the code changed to call `iglob` 10 times repeatedly in the future... as long as it still works, and is there for another reason, it shouldn't affect the passing of this test). Shall I remove them @lorenabalan?